packages/openapi-ts/src/utils/write/types.ts

```diff
@@ -259,15 +259,14 @@ const processServiceTypes = ({
       methodMap.$ref = operation.name;
 
       if (hasReq) {
-        const bodyParameter = operation.parameters
-          .filter((parameter) => parameter.in === 'body')
-          .sort(sorterByName)[0];
+        const bodyParameter = operation.parametersBody;
```

Comment:
Did you find out why these parameters aren't marked as `in: 'body'`?

Reply:
As I said in the bug-report - in some cases the request body (if it is from type 'application/x-www-form-urlencoded' or 'multipart/form-data') will be transformed to formData type parameter. On why it is this way - I think because the old clients have the special handling for formData in https://github.com/hey-api/openapi-ts/blob/a2cb1b06b4a7c7bdcac36400a26a7e19fca15bd2/packages/openapi-ts/src/templates/core/functions/getFormData.hbs and the logic there is, if there is an existing body it will be sent as it is https://github.com/hey-api/openapi-ts/blob/a2cb1b06b4a7c7bdcac36400a26a7e19fca15bd2/packages/openapi-ts/src/templates/core/fetch/sendRequest.hbs#L14, and if there is no body, but there is a form data, it will be transformed in the getFormData and sent as body. The new clients do not have this differentiated handling for body and formData so the formData is completely missing from the new clients. As I said in the bug, this fix will probably still miss the real formData fields and they probably need to be added in some way.

Reply:
Are you able to add an example/test case for "real formData" fields?

Reply:
In fact there is an example test, which does not work right now ->
```json
    "/api/v{api-version}/upload": {
      "post": {
        "tags": ["Upload"],
        "operationId": "UploadFile",
        "parameters": [
          {
            "description": "Supply a file reference for upload",
            "name": "file",
            "in": "formData",
            "required": true,
            "schema": {
              "type": "file"
            }
          },
          {
            "name": "api-version",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string",
              "nullable": true
            }
          }
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "type": "boolean"
                }
              }
            }
          }
        }
      }
    },
```
The generated code right now is services.gen.ts in the new client-fetch test:
```typescript
export const uploadFile = (options: Options<UploadFileData>) => {
  return (options?.client ?? client).post<UploadFileResponse, UploadFileError>({
    ...options,
    url: '/api/v{api-version}/upload',
  });
};
```
and types.gen.ts in the new client-fetch test:
```typescript
export type UploadFileData = unknown;

export type UploadFileResponse = boolean;

export type UploadFileError = unknown;
```

Here is the services.gen.ts in the old axis code:
```typescript
  public static uploadFile(
    data: UploadFileData,
  ): CancelablePromise<UploadFileResponse> {
    return __request(OpenAPI, {
      method: 'POST',
      url: '/api/v{api-version}/upload',
      formData: {
        file: data.file,
      },
    });
  }
}
```
and the types.gen.ts agin in the axis code:
```typescript

export type UploadFileData = {
  /**
   * Supply a file reference for upload
   */
  file: Blob | File;
};

export type UploadFileResponse = boolean;
```
I'll try to fix it direct in this PullRequest.

Reply:
Thank you!

Reply:
It seems that the fix will not be that easy - I'll write a bug request for it and try to fix it in another pull request. The problem is that the real formdata fields are not created through getOperationRequestBody and as such do not have all fields, so they need to be handled somewhat differently.